plugins/query-insights/src/test/java/org/opensearch/plugin/insights/rules/action/top_queries/TopQueriesTests.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.insights.rules.action.top_queries;
+
+import org.opensearch.common.io.stream.BytesStreamOutput;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.plugin.insights.QueryInsightsTestUtils;
+import org.opensearch.test.OpenSearchTestCase;
+
+import java.io.IOException;
+
+/**
+ * Tests for {@link TopQueries}.
+ */
+public class TopQueriesTests extends OpenSearchTestCase {
```

Comment:
What if we have a case with multiple queries all taking the same time and we need to evict some of these from the window. Which ones do we evict in this case? @ansjcy

Reply:
The PriorityQueue would evict either one of them. But we can make some follow ups to evaluate the best way to handle this scenario.